src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    private static void loadTasks() {
+        try {
+            addTasksIntoList();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    private static File loadFile() throws IOException {
+        File file = new File(FILE_PATH);
+        if (!file.exists()) {
+            File dir = new File("data");
+            dir.mkdir();
+            File newFile = new File("data/duke.txt");
+            newFile.createNewFile();
+            return newFile;
+        }
+        return file;
+    }
+
+    private static void addTasksIntoList() throws IOException {
+        File f = loadFile();
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String entry = s.nextLine();
+            String[] entryComponents = entry.split(GAP);
+            String description = entryComponents[0] + " " + entryComponents[2];
+            switch (entryComponents[0]) {
+            case "T":
+                tasks.add(new Todo(description));
+                break;
+            case "D":
+                String by = "by " + entryComponents[3];
+                tasks.add(new Deadline(description, by));
+                break;
+            case "E":
+                String at = "at " + entryComponents[3];
+                tasks.add(new Event(description, at));
+                break;
+            default:
+                break;
+            }
+            if (entryComponents[1].equals("X")) {
+                tasks.get(tasks.size() - 1).completeTask();;
+            }
+        }
+    }
+
+    private static void saveTasks() {
+        try {
+            writeTasksToFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+    }
+
+    private static void writeTasksToFile() throws IOException {
+        FileWriter fw = new FileWriter(FILE_PATH);
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            String details = task.getDescription().substring(task.getDescription().indexOf(" ") + 1);
+            String date = "";
+            if (task instanceof Deadline) {
+                date = ((Deadline) task).getBy().split(" ", 2)[1].trim();
+            } else if (task instanceof Event) {
+                date = ((Event) task).getAt().split(" ", 2)[1].trim();
+            }
+            String taskLabel = task.getType() + GAP + task.getIsDone() + GAP;
+            String taskBody = details.trim() + GAP + date + System.lineSeparator();
+            fw.write(taskLabel + taskBody);
+        }
+        fw.close();
+    }
+
+    private static void handleCommand() {
+        String input;
+        Scanner in = new Scanner(System.in);
+        input = in.nextLine();
+        while (!input.equals("bye")) {
+            try {
+                String keyword = input.split(" ")[0].toLowerCase();
+                switch (keyword) {
+                case "bye":
```

Comment:
When the user types the "bye" command with some extensions (so the command is not matching the "bye" String in the condition of the while loop), isn't the user still want to close the program? Or maybe we should prompt something to the user instead of ignoring it?